pkg/destroy/aws/aws.go

```diff
@@ -185,19 +185,25 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) ([]string, erro
 	// running on the cluster creating new resources while we are attempting to delete resources, which could leak
 	// the new resources.
 	ec2Client := ec2.New(awsSession)
+	shutdownTimeout := time.Now()
 	err = wait.PollImmediateUntil(
 		time.Second*10,
 		func() (done bool, err error) {
-			instancesToDelete, err := o.findEC2Instances(ctx, ec2Client, deleted)
+			instancesRunning, instancesNotTerminated, err := o.findEC2Instances(ctx, ec2Client, deleted)
 			if err != nil {
 				o.Logger.WithError(err).Info("error while finding EC2 instances to delete")
 				if err := ctx.Err(); err != nil {
 					return false, err
 				}
 			}
-			if len(instancesToDelete) == 0 && err == nil {
+			if len(instancesNotTerminated) == 0 && len(instancesRunning) == 0 && err == nil {
 				return true, nil
 			}
+			instancesToDelete := instancesRunning
+			if time.Since(shutdownTimeout) > 5*time.Second {
+				instancesToDelete = instancesNotTerminated
+				shutdownTimeout = time.Now().Add(5 * time.Second)
```

Comment:
You don't want to add 5 seconds. You are checking that it has been more than 5 seconds since `shutdownTimeout` above, so adding 5 seconds here effectively makes the check that it has been more than 10 seconds since the last mass terminate.
```suggestion
				shutdownTimeout = time.Now()
```

Reply:
If I don't add 5 seconds here, it would keep trying to mass terminate after 5 seconds endlessly. I was hoping to request just one more mass terminate in the 10 second timeout allotted for the destroy instances.

Reply:
oh wait no nvm I'l fix this